### DISCUSSION TITLE
Periscope / Operate Guns order icon improvements & modding.

### DISCUSSION THREAD

**Discussion by Ek-Videogames** [FakeFishGames/Barotrauma#5284] [Ideas and suggestions]
Positioning of icons when giving operate weapon commands can sometimes be not very obvious or clear on larger submarines or if two turrets are overlapping each other. This could be improved by some simple method such as having a line pointing from the icon to the turret's true position on the minimap.
![image](https://user-images.githubusercontent.com/55356135/111912867-e9813b00-8a41-11eb-88ad-f4df7c50e6bd.png)

All other turrets besides coilgun and railgun, such as new ones or reskins added by mods, share the same generic gun icon. I'm suggesting a component for defining a custom icon appearing in the gun orders screen, for example something like a
 `<TurretIcon texture="customicon.png" sourcerect="0,0,64,64" />` .

![image](https://user-images.githubusercontent.com/55356135/111913287-db341e80-8a43-11eb-959b-c23e2318a627.png)

**Comment by ekqvist** [comment]
You've probably already noticed, but these ideas have been implemented in the game. We draw lines to better display the positions of the weapons and the weapon icons can now be directly defined in the item definitions.
